cvat/apps/engine/views.py

```diff
@@ -851,16 +859,179 @@ def jobs(self, request, pk):
 
         return Response(serializer.data)
 
+    def _is_data_uploading(self) -> bool:
+        return 'data' in self.action
+
     # UploadMixin method
     def get_upload_dir(self):
         if 'annotations' in self.action:
             return self._object.get_tmp_dirname()
-        elif 'data' in self.action:
+        elif self._is_data_uploading():
             return self._object.data.get_upload_dirname()
         elif 'backup' in self.action:
             return backup.get_backup_dirname()
         return ""
 
+    def _get_tus_input_ordering_metafile_name(self) -> str:
+        return "__tus_input_order__"
+
+    def _get_tus_custom_ordering_metafile_name(self) -> str:
+        return "__tus_custom_order__"
+
+    def _get_tus_metafile_names(self) -> List[str]:
+        return [
+            self._get_tus_input_ordering_metafile_name(),
+            self._get_tus_custom_ordering_metafile_name(),
+        ]
+
+    # UploadMixin method
+    def validate_uploaded_file_name(self, filename: str) -> bool:
+        return (
+            super().validate_uploaded_file_name(filename) and
+            osp.basename(filename) not in self._get_tus_metafile_names()
```

Comment:
IMO, we should just reserve some subset of filenames (e.g. all names beginning with `__cvat`) for internal use, and reject them in the base class's implementation of `validate_uploaded_file_name`. That way, we won't risk colliding with a file name already being used by someone if we need to add more special names later.

Reply:
Done.